// native-base/lib/commonjs/components/composites/Menu/MenuItemOption.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuItemOption = void 0;

var _react = _interopRequireDefault(require("react"));

var _primitives = require("../../primitives");

var _hooks = require("../../../hooks");

var _MenuItem = require("./MenuItem");

var _MenuOptionGroup = require("./MenuOptionGroup");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const MenuItemOption = /*#__PURE__*/_react.default.memo(({
  value,
  children,
  onPress,
  ...props
}) => {
  const {
    values,
    onChange,
    type
  } = _react.default.useContext(_MenuOptionGroup.MenuOptionContext);

  const modifiedOnPress = e => {
    onChange(value);
    onPress && onPress(e);
  };

  let iconName = type === 'checkbox' ? values.includes(value) ? 'check-box' : 'check-box-outline-blank' : values.includes(value) ? 'radio-button-checked' : 'radio-button-unchecked';
  const newProps = (0, _hooks.useThemeProps)('MenuItem', props);
  return /*#__PURE__*/_react.default.createElement(_MenuItem.MenuItem, _extends({}, props, {
    px: 0,
    py: 0,
    onPress: modifiedOnPress
  }), /*#__PURE__*/_react.default.createElement(_primitives.Flex, {
    direction: "row",
    px: newProps.px,
    py: newProps.py
  }, /*#__PURE__*/_react.default.createElement(_primitives.Icon, {
    name: iconName,
    pr: newProps.px,
    size: newProps._text.fontSize
  }), children));
});

exports.MenuItemOption = MenuItemOption;
