// native-base/lib/commonjs/components/composites/Menu/Menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Menu = void 0;

var _react = _interopRequireDefault(require("react"));

var _View = _interopRequireDefault(require("../../primitives/View"));

var _hooks = require("../../../hooks");

var _core = require("../../../core");

var _focus = require("@react-native-aria/focus");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Menu = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(({
  trigger,
  closeOnSelect = true,
  children,
  onOpen,
  onClose,
  ...props
}, ref) => {
  let triggerRef = _react.default.useRef();

  const newProps = (0, _hooks.useThemeProps)('Menu', props);

  let [isOpen, toggle] = _react.default.useState(false);

  const {
    setPopover,
    closePopover
  } = (0, _core.usePopover)();

  const closeMenu = () => {
    closePopover();
    toggle(false);
    onClose && onClose();
  };

  const openMenu = () => {
    setPopover( /*#__PURE__*/_react.default.createElement(_View.default, _extends({}, newProps, {
      ref: ref
    }), /*#__PURE__*/_react.default.createElement(_focus.FocusScope, {
      restoreFocus: true,
      autoFocus: true
    }, children)), {
      triggerRef,
      animationDuration: 200,
      onClose: closeMenu,
      parentComponentConfig: {
        open: isOpen,
        closeMenu,
        closeOnSelect
      }
    });
    toggle(true);
    onOpen && onOpen();
  };

  return trigger({
    onPress: openMenu,
    ref: triggerRef
  }, {
    open: isOpen
  });
}));

exports.Menu = Menu;
