// native-base/lib/module/components/composites/Avatar/Avatar.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { Box, Image, Text } from '../../primitives';
import { useThemeProps } from '../../../hooks';

const Avatar = (props, ref) => {
  const [error, setError] = React.useState(false);
  const {
    size,
    style,
    source,
    children,
    ...remainingProps
  } = props;
  const {
    _text,
    ...newProps
  } = useThemeProps('Avatar', { ...remainingProps,
    name: 'avatar',
    size
  });
  let Badge = /*#__PURE__*/React.createElement(React.Fragment, null);
  let remainingChildren = []; //  Pop Badge from children

  React.Children.map(children, (child, key) => {
    var _child$type$type;

    if (typeof child.type === 'object' && ((_child$type$type = child.type.type) === null || _child$type$type === void 0 ? void 0 : _child$type$type.name) === 'AvatarBadge') {
      Badge = child;
    } else {
      remainingChildren.push(typeof child === 'string' ? /*#__PURE__*/React.createElement(Text, _extends({
        key: 'avatar-children-' + key
      }, _text), child) : child);
    }
  });
  const imageFitStyle = {
    height: '100%',
    width: '100%'
  };
  return /*#__PURE__*/React.createElement(Box, _extends({}, newProps, {
    style: style,
    ref: ref
  }), source && !error ? /*#__PURE__*/React.createElement(Image, {
    borderRadius: newProps.borderRadius,
    source: source,
    alt: '--',
    _alt: _text,
    style: [style, imageFitStyle],
    onError: () => {
      setError(true);
    }
  }) : remainingChildren.length === 0 ? /*#__PURE__*/React.createElement(Text, _text, "--") // default alternate
  : remainingChildren, Badge);
};

export default /*#__PURE__*/React.memo( /*#__PURE__*/React.forwardRef(Avatar));
