// native-base/lib/commonjs/core/color-mode/hooks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useColorModeValue = useColorModeValue;
exports.useModeManager = useModeManager;
exports.useColorMode = void 0;

var _react = _interopRequireWildcard(require("react"));

var _Context = require("./../hybrid-overlay/Context");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const useColorMode = () => {
  const {
    colorMode: context
  } = _react.default.useContext(_Context.HybridContext);

  if (context === undefined) {
    throw new Error('useColorMode must be used within a NativeBaseProvider');
  }

  return context;
};

exports.useColorMode = useColorMode;

function useColorModeValue(light, dark) {
  const {
    colorMode
  } = useColorMode();
  return colorMode === 'dark' ? dark : light;
}

function useModeManager(initialColorMode, colorModeManager) {
  const [colorMode, setRawMode] = (0, _react.useState)(initialColorMode);

  async function setColorMode(val) {
    if (colorModeManager) {
      await colorModeManager.set(val);
    }

    setRawMode(val);
  } // For initial setting initial color mode from storage


  (0, _react.useEffect)(() => {
    if (colorModeManager) {
      (async function getMode() {
        let value = await colorModeManager.get(initialColorMode);

        if (value && value !== colorMode) {
          setRawMode(value);
        }
      })();
    }
  }, [colorMode, initialColorMode, colorModeManager]);
  return {
    colorMode,
    setColorMode
  };
}
