// native-base/lib/commonjs/components/primitives/Flex/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Spacer = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("../Box"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Flex = ({
  style,
  direction,
  align,
  justify,
  wrap,
  basis,
  grow,
  shrink,
  ...props
}, ref) => {
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({}, props, {
    display: "flex",
    flexDirection: direction || 'column',
    alignItems: align,
    justifyContent: justify,
    flexGrow: grow,
    flexBasis: basis,
    flexShrink: shrink,
    flexWrap: wrap,
    style: style,
    ref: ref
  }));
}; //Spacer Component that adds space between components where it is placed


const Spacer = props => {
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({
    flexGrow: 1
  }, props));
};

exports.Spacer = Spacer;

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(Flex));

exports.default = _default;
