// native-base/lib/commonjs/components/composites/NumberInput/NumberInputStepper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NBStepper = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _primitives = require("../../primitives");

var _hooks = require("../../../hooks");

var _Context = require("./Context");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const NBStepper = ({
  children,
  ...props
}) => {
  const {
    style,
    isIncrement,
    disablitityCheck,
    _active,
    _disabled,
    isDisabled,
    ariaLabel,
    pressHandler,
    iconColor,
    ...newProps
  } = (0, _hooks.useThemeProps)('NumberInputStepper', props);
  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    activeOpacity: 1,
    disabled: disablitityCheck || isDisabled,
    onPress: pressHandler,
    accessible: true,
    accessibilityLabel: ariaLabel
  }, /*#__PURE__*/_react.default.createElement(_primitives.Box, _extends({}, newProps, _active, disablitityCheck || isDisabled ? _disabled : {}, {
    borderColor: "transparent",
    style: style,
    opacity: disablitityCheck || isDisabled ? 0.4 : 1
  }, _reactNative.Platform.OS === 'web' ? {
    disabled: disablitityCheck || isDisabled,
    cursor: disablitityCheck || isDisabled ? 'not-allowed' : 'auto'
  } : {}), children || isIncrement ? /*#__PURE__*/_react.default.createElement(_primitives.Icon, {
    name: "arrow-drop-up",
    type: "MaterialIcons",
    color: iconColor
  }) : /*#__PURE__*/_react.default.createElement(_primitives.Icon, {
    name: "arrow-drop-down",
    type: "MaterialIcons",
    color: iconColor
  })));
};

exports.NBStepper = NBStepper;

const NumberInputStepper = ({
  children,
  ...props
}) => {
  const {
    numberInputStepper,
    setNumberInputStepper
  } = _react.default.useContext(_Context.NumberInputContext);

  _react.default.useEffect(() => {
    !numberInputStepper && setNumberInputStepper( /*#__PURE__*/_react.default.createElement(_primitives.VStack, props, children));
  }, [numberInputStepper, setNumberInputStepper, props, children]);

  return null;
};

var _default = /*#__PURE__*/_react.default.memo(NumberInputStepper);

exports.default = _default;
