// native-base/lib/module/theme/components/alert.js
import { mode, getColor, getColorScheme, transparentize } from '../tools';

function getBg(props) {
  let {
    theme,
    status,
    variant
  } = props;
  status = getColorScheme(props, status);
  const lightBg = variant === 'solid' ? getColor(theme, "".concat(status, ".400"), status) : getColor(theme, "".concat(status, ".100"), status);
  const darkBg = transparentize("".concat(status, ".500"), variant === 'solid' ? 0.8 : 0.5)(theme);
  return mode(lightBg, darkBg)(props);
}

const variantSubtle = props => {
  let {
    status,
    theme
  } = props;
  status = getColorScheme(props, status);
  return {
    bg: getBg(props),
    iconColor: mode("".concat(status, ".500"), "".concat(status, ".200"))(props),
    textColor: mode(getColor(theme, "".concat(status, ".600"), status), getColor(theme, "".concat(status, ".300"), status))(props)
  };
};

const variantOutline = props => {
  let {
    status,
    theme
  } = props;
  status = getColorScheme(props, status);
  return {
    borderWidth: 1,
    borderColor: mode("".concat(status, ".600"), "".concat(status, ".500"))(props),
    iconColor: mode("".concat(status, ".500"), "".concat(status, ".500"))(props),
    textColor: getColor(theme, "".concat(status, ".700"), status)
  };
};

const variantOutlineLight = props => {
  let {
    status,
    theme
  } = props;
  status = getColorScheme(props, status);
  return {
    borderWidth: 1,
    borderColor: transparentize("".concat(status, ".600"), 0.2)(theme),
    iconColor: mode("".concat(status, ".500"), "".concat(status, ".200"))(props),
    textColor: getColor(theme, "".concat(status, ".900"), status)
  };
};

const variantSolid = props => {
  let {
    status
  } = props;
  status = getColorScheme(props, status);
  return {
    borderWidth: 6,
    borderColor: 'transparent',
    bg: getBg(props),
    iconColor: mode("white", "".concat(status, ".300"))(props),
    textColor: mode('white', "".concat(status, ".200"))(props)
  };
};

const variantLeftAccent = props => {
  let {
    status
  } = props;
  status = getColorScheme(props, status);
  return {
    borderWidth: 4,
    bg: getBg(props),
    iconColor: mode("".concat(status, ".500"), "".concat(status, ".200"))(props),
    textColor: mode("".concat(status, ".600"), "".concat(status, ".300"))(props),
    borderColor: 'transparent',
    borderLeftColor: mode("".concat(status, ".600"), "".concat(status, ".300"))(props)
  };
};

const variantTopAccent = props => {
  let {
    status
  } = props;
  status = getColorScheme(props, status);
  return {
    borderWidth: 4,
    borderColor: 'transparent',
    borderTopColor: mode("".concat(status, ".600"), "".concat(status, ".300"))(props),
    bg: getBg(props),
    iconColor: mode("".concat(status, ".500"), "".concat(status, ".200"))(props),
    textColor: mode("".concat(status, ".600"), "".concat(status, ".300"))(props)
  };
};

const variants = {
  'subtle': variantSubtle,
  'solid': variantSolid,
  'left-accent': variantLeftAccent,
  'top-accent': variantTopAccent,
  'outline': variantOutline,
  'outline-light': variantOutlineLight
};
export const Alert = {
  defaultProps: {
    status: 'default',
    variant: 'subtle',
    p: 4,
    borderRadius: 'lg'
  },
  variants
}; // AlertTitle

export const AlertTitle = {
  defaultProps: {
    fontSize: 'md',
    fontWeight: 'semibold'
  }
}; // AlertDescription

export const AlertDescription = {
  defaultProps: {
    fontSize: 'sm',
    fontWeight: 'medium'
  }
};
