// native-base/lib/module/theme/components/badge.js
import { mode, getColorScheme } from '../tools';
const baseStyle = {
  px: 1,
  py: 1,
  _text: {
    fontSize: 'xs',
    textTransform: 'uppercase',
    fontWeight: 'bold'
  },
  borderRadius: 'lg'
};

function variantSolid(props) {
  let colorScheme = getColorScheme(props);
  return {
    bg: mode("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props),
    borderWidth: 1,
    borderColor: 'transparent'
  };
}

function variantSubtle(props) {
  let colorScheme = getColorScheme(props);
  return {
    bg: mode("".concat(colorScheme, ".100"), "".concat(colorScheme, ".500"))(props),
    _text: {
      color: mode("".concat(colorScheme, ".700"), "".concat(colorScheme, ".100"))(props)
    },
    borderWidth: 1,
    borderColor: 'transparent'
  };
}

function variantOutline(props) {
  let colorScheme = getColorScheme(props);
  return {
    borderColor: mode("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props),
    _text: {
      color: mode("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props)
    },
    borderWidth: 1
  };
}

const variants = {
  solid: variantSolid,
  subtle: variantSubtle,
  outline: variantOutline
};
const defaultProps = {
  variant: 'subtle',
  colorScheme: 'gray'
};
export default {
  baseStyle,
  variants,
  defaultProps
};
