// native-base/lib/commonjs/components/composites/Fab/Fab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _utils = require("../../../theme/tools/utils");

var _Button = _interopRequireDefault(require("../../primitives/Button"));

var _Stack = require("../../primitives/Stack");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Fab = (props, ref) => {
  const {
    label,
    icon,
    placement = 'bottom-right',
    ...newProps
  } = (0, _utils.omitUndefined)(props);
  const placementProps = {
    'top-right': {
      top: 4,
      right: 4,
      position: 'absolute'
    },
    'top-left': {
      top: 4,
      left: 4,
      position: 'absolute'
    },
    'bottom-right': {
      bottom: 4,
      right: 4,
      position: 'absolute'
    },
    'bottom-left': {
      bottom: 4,
      left: 4,
      position: 'absolute'
    }
  };
  return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
    variant: "unstyled",
    bg: "default.200",
    rounded: "full",
    zIndex: 20,
    p: 2
  }, placementProps[placement], newProps, {
    ref: ref
  }), icon && label ? /*#__PURE__*/_react.default.createElement(_Stack.HStack, {
    space: 1,
    justifyContent: "center",
    alignItems: "center"
  }, icon, label) : icon !== null && icon !== void 0 ? icon : label);
};

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(Fab));

exports.default = _default;
