// native-base/lib/commonjs/components/primitives/Input/InputGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputGroup = exports.InputRightAddon = exports.InputLeftAddon = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("../Box"));

var _utils = require("../../../utils");

var _Flex = _interopRequireDefault(require("../Flex"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const addonsDefaultStyle = {
  p: 3,
  borderColor: 'gray.300',
  borderWidth: 1
};

const InputLeftAddon = /*#__PURE__*/_react.default.memo(props => {
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({}, addonsDefaultStyle, {
    borderRightWidth: 0,
    roundedLeft: 4,
    bg: "gray.200"
  }, props), /*#__PURE__*/_react.default.createElement(_Box.default, {
    m: "auto"
  }, props.children));
});

exports.InputLeftAddon = InputLeftAddon;

const InputRightAddon = /*#__PURE__*/_react.default.memo(props => {
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({}, addonsDefaultStyle, {
    borderLeftWidth: 0,
    roundedRight: 4,
    bg: "gray.200"
  }, props), /*#__PURE__*/_react.default.createElement(_Box.default, {
    m: "auto"
  }, props.children));
});

exports.InputRightAddon = InputRightAddon;

const supplyPropsToChildren = (children, props) => {
  return _react.default.Children.map(children, child => {
    return /*#__PURE__*/_react.default.cloneElement(child, props, child.props.children);
  });
};

const InputGroup = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(({
  children,
  ...props
}, ref) => {
  return /*#__PURE__*/_react.default.createElement(_Flex.default, {
    direction: "row",
    ref: ref
  }, supplyPropsToChildren((0, _utils.getAttachedChildren)(children), props));
}));

exports.InputGroup = InputGroup;
