// native-base/lib/module/components/composites/Stat/index.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { Box, Text, Icon, HStack } from '../../primitives';
import { useThemeProps } from '../../../hooks';
export const StatLabel = /*#__PURE__*/React.memo(({
  style,
  ...props
}) => {
  let newProps = useThemeProps('Stat', props);
  return /*#__PURE__*/React.createElement(Text, _extends({}, newProps._statLabel, newProps, {
    style: style
  }, props), props.children);
});
export const StatNumber = /*#__PURE__*/React.memo(({
  style,
  ...props
}) => {
  let newProps = useThemeProps('Stat', props);
  return /*#__PURE__*/React.createElement(Text, _extends({}, newProps._statNumber, newProps, {
    style: style
  }), props.children);
});
export const StatHelpText = /*#__PURE__*/React.memo(({
  style,
  ...props
}) => {
  let newProps = useThemeProps('Stat', props);
  return /*#__PURE__*/React.createElement(Box, _extends({}, newProps._statHelpText, newProps, {
    style: style
  }), props.children);
});
export const StatArrow = /*#__PURE__*/React.memo(({
  type,
  ...props
}) => {
  return /*#__PURE__*/React.createElement(Icon, _extends({
    ml: -1,
    type: "Entypo",
    name: type === 'increase' ? 'triangle-up' : 'triangle-down',
    color: type === 'increase' ? 'green.500' : 'red.500',
    size: 8
  }, props));
});
export const StatGroup = /*#__PURE__*/React.memo( /*#__PURE__*/React.forwardRef(({
  style,
  ...props
}, ref) => {
  let newProps = useThemeProps('Stat', props);
  return /*#__PURE__*/React.createElement(HStack, _extends({}, newProps._statGroup, newProps, {
    ref: ref,
    style: style
  }));
}));
const StatMain = /*#__PURE__*/React.forwardRef(({
  style,
  ...props
}, ref) => {
  return /*#__PURE__*/React.createElement(Box, _extends({}, props, {
    ref: ref,
    style: style
  }));
});
const StatTemp = StatMain;
StatTemp.Label = StatLabel;
StatTemp.Number = StatNumber;
StatTemp.HelpText = StatHelpText;
StatTemp.Arrow = StatArrow;
StatTemp.Group = StatGroup;
const Stat = StatTemp;
export default Stat;
