// native-base/lib/commonjs/components/composites/Modal/ModalOverlay.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _Box = _interopRequireDefault(require("../../primitives/Box"));

var _Context = require("./Context");

var _hooks = require("../../../hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ModalOverlay = ({
  children,
  ...props
}) => {
  const newProps = (0, _hooks.useThemeProps)('ModalOverlay', props);

  const {
    toggleVisible,
    toggleOnClose
  } = _react.default.useContext(_Context.ModalContext);

  return /*#__PURE__*/_react.default.createElement(_Box.default, newProps, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    style: {
      position: 'absolute',
      left: 0,
      top: 0,
      right: 0,
      bottom: 0
    },
    accessible: false,
    onPress: newProps.closeOnOverlayClick === false ? () => {} : () => {
      toggleVisible(false);
      toggleOnClose(false);
    }
  }), children);
};

var _default = /*#__PURE__*/_react.default.memo(ModalOverlay);

exports.default = _default;
