// native-base/lib/commonjs/components/composites/Progress/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _primitives = require("../../primitives");

var _hooks = require("../../../hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Progress = ({
  value,
  isIndeterminate,
  ...props
}, ref) => {
  // const width = new Animated.Value(0);
  // useEffect(() => {
  //   Animated.loop(
  //     Animated.timing(width, {
  //       toValue: 270,
  //       duration: 1000,
  //       useNativeDriver: true,
  //     })
  //   ).start();
  // });
  let newProps = (0, _hooks.useThemeProps)('Progress', props);
  const {
    innerBg
  } = newProps;
  const innerProps = {
    bg: innerBg,
    shadow: 0,
    rounded: newProps.rounded,
    height: '100%',
    w: value + '%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: 'white'
  };
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, _extends({}, newProps, {
    style: props.style,
    ref: ref
  }), isIndeterminate ?
  /*#__PURE__*/
  // <Animated.View
  //   style={[
  //     {
  //       transform: [
  //         {
  //           translateX: width,
  //         },
  //       ],
  //     },
  //   ]}
  // >
  // </Animated.View>
  _react.default.createElement(_primitives.Box, _extends({}, innerProps, {
    children: newProps.children
  })) : /*#__PURE__*/_react.default.createElement(_primitives.Box, _extends({}, innerProps, {
    children: newProps.children
  })));
};

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(Progress));

exports.default = _default;
