// native-base/lib/module/components/primitives/Slider/SliderThumb.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { StyleSheet, Animated, Platform } from 'react-native';
import { SliderContext } from './Context';
import Box from '../Box';
import Icon from '../Icon';
import { useThemeProps } from '../../../hooks';

const SliderThumb = ({
  children,
  ...props
}) => {
  const { ...newProps
  } = useThemeProps('SliderThumb', props);
  const {
    sliderOffset = 0,
    panResponder,
    colorScheme,
    thumbSize = 0,
    orientation,
    isDisabled
  } = React.useContext(SliderContext);
  const sliderThumbPosition = sliderOffset - 8 - thumbSize / 2;
  const customStyle = StyleSheet.create({
    SliderThumb: {
      position: 'absolute',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 999,
      padding: 4 // increased touch area for better touch detection.

    },
    verticalStyle: {
      bottom: sliderThumbPosition
    },
    horizontalStyle: {
      left: sliderThumbPosition
    }
  });

  const sizedIcon = () => /*#__PURE__*/React.cloneElement(children, {
    size: "".concat(thumbSize, "px"),
    color: children.props.color ? children.props.color : colorScheme
  }, children.props.children);

  return /*#__PURE__*/React.createElement(Animated.View, _extends({
    style: [customStyle.SliderThumb, orientation === 'vertical' ? customStyle.verticalStyle : customStyle.horizontalStyle]
  }, panResponder.panHandlers), /*#__PURE__*/React.createElement(Box, _extends({
    cursor: Platform.OS === 'web' ? isDisabled ? 'not-allowed' : 'pointer' : null,
    position: "relative",
    borderRadius: "full",
    backgroundColor: "light.50",
    p: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    userSelect: Platform.OS === 'web' ? 'none' : null
  }, newProps), children ? sizedIcon() : /*#__PURE__*/React.createElement(Icon, {
    name: "circle-medium",
    type: "MaterialCommunityIcons",
    color: colorScheme,
    size: "".concat(thumbSize, "px"),
    opacity: 0
  })));
};

export default SliderThumb;
