// native-base/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  AppBar: true,
  Badge: true,
  IconButton: true,
  Alert: true,
  AspectRatio: true,
  Avatar: true,
  Breadcrumb: true,
  CloseButton: true,
  Container: true,
  Divider: true,
  Kbd: true,
  Progress: true,
  Accordion: true,
  Skeleton: true,
  FormControl: true,
  Stat: true,
  Tag: true,
  Code: true,
  Center: true,
  Square: true,
  Circle: true,
  Wrap: true,
  PinInput: true,
  Fade: true,
  useFadeTransition: true,
  ScaleFade: true,
  Slide: true,
  SlideFade: true,
  NumberInput: true,
  NumberInputField: true,
  NumberInputStepper: true,
  NumberIncrementStepper: true,
  NumberDecrementStepper: true,
  Collapse: true,
  CircularProgress: true,
  Modal: true,
  Popover: true,
  AlertDialog: true,
  useToast: true,
  Menu: true,
  SimpleGrid: true,
  Tabs: true,
  Actionsheet: true,
  Snackbar: true,
  useSnackbar: true,
  Fab: true,
  Typeahead: true,
  useTypeahead: true,
  View: true,
  Text: true,
  Checkbox: true,
  Radio: true,
  Button: true,
  ButtonGroup: true,
  Column: true,
  Row: true,
  Box: true,
  Stack: true,
  VStack: true,
  HStack: true,
  ZStack: true,
  Slider: true,
  Icon: true,
  createIcon: true,
  Input: true,
  InputGroup: true,
  InputLeftAddon: true,
  InputRightAddon: true,
  Image: true,
  Spinner: true,
  Heading: true,
  Flex: true,
  Switch: true,
  TextArea: true,
  Link: true,
  Spacer: true,
  List: true,
  Select: true
};
Object.defineProperty(exports, "AppBar", {
  enumerable: true,
  get: function () {
    return _composites.AppBar;
  }
});
Object.defineProperty(exports, "Badge", {
  enumerable: true,
  get: function () {
    return _composites.Badge;
  }
});
Object.defineProperty(exports, "IconButton", {
  enumerable: true,
  get: function () {
    return _composites.IconButton;
  }
});
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function () {
    return _composites.Alert;
  }
});
Object.defineProperty(exports, "AspectRatio", {
  enumerable: true,
  get: function () {
    return _composites.AspectRatio;
  }
});
Object.defineProperty(exports, "Avatar", {
  enumerable: true,
  get: function () {
    return _composites.Avatar;
  }
});
Object.defineProperty(exports, "Breadcrumb", {
  enumerable: true,
  get: function () {
    return _composites.Breadcrumb;
  }
});
Object.defineProperty(exports, "CloseButton", {
  enumerable: true,
  get: function () {
    return _composites.CloseButton;
  }
});
Object.defineProperty(exports, "Container", {
  enumerable: true,
  get: function () {
    return _composites.Container;
  }
});
Object.defineProperty(exports, "Divider", {
  enumerable: true,
  get: function () {
    return _composites.Divider;
  }
});
Object.defineProperty(exports, "Kbd", {
  enumerable: true,
  get: function () {
    return _composites.Kbd;
  }
});
Object.defineProperty(exports, "Progress", {
  enumerable: true,
  get: function () {
    return _composites.Progress;
  }
});
Object.defineProperty(exports, "Accordion", {
  enumerable: true,
  get: function () {
    return _composites.Accordion;
  }
});
Object.defineProperty(exports, "Skeleton", {
  enumerable: true,
  get: function () {
    return _composites.Skeleton;
  }
});
Object.defineProperty(exports, "FormControl", {
  enumerable: true,
  get: function () {
    return _composites.FormControl;
  }
});
Object.defineProperty(exports, "Stat", {
  enumerable: true,
  get: function () {
    return _composites.Stat;
  }
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function () {
    return _composites.Tag;
  }
});
Object.defineProperty(exports, "Code", {
  enumerable: true,
  get: function () {
    return _composites.Code;
  }
});
Object.defineProperty(exports, "Center", {
  enumerable: true,
  get: function () {
    return _composites.Center;
  }
});
Object.defineProperty(exports, "Square", {
  enumerable: true,
  get: function () {
    return _composites.Square;
  }
});
Object.defineProperty(exports, "Circle", {
  enumerable: true,
  get: function () {
    return _composites.Circle;
  }
});
Object.defineProperty(exports, "Wrap", {
  enumerable: true,
  get: function () {
    return _composites.Wrap;
  }
});
Object.defineProperty(exports, "PinInput", {
  enumerable: true,
  get: function () {
    return _composites.PinInput;
  }
});
Object.defineProperty(exports, "Fade", {
  enumerable: true,
  get: function () {
    return _composites.Fade;
  }
});
Object.defineProperty(exports, "useFadeTransition", {
  enumerable: true,
  get: function () {
    return _composites.useFadeTransition;
  }
});
Object.defineProperty(exports, "ScaleFade", {
  enumerable: true,
  get: function () {
    return _composites.ScaleFade;
  }
});
Object.defineProperty(exports, "Slide", {
  enumerable: true,
  get: function () {
    return _composites.Slide;
  }
});
Object.defineProperty(exports, "SlideFade", {
  enumerable: true,
  get: function () {
    return _composites.SlideFade;
  }
});
Object.defineProperty(exports, "NumberInput", {
  enumerable: true,
  get: function () {
    return _composites.NumberInput;
  }
});
Object.defineProperty(exports, "NumberInputField", {
  enumerable: true,
  get: function () {
    return _composites.NumberInputField;
  }
});
Object.defineProperty(exports, "NumberInputStepper", {
  enumerable: true,
  get: function () {
    return _composites.NumberInputStepper;
  }
});
Object.defineProperty(exports, "NumberIncrementStepper", {
  enumerable: true,
  get: function () {
    return _composites.NumberIncrementStepper;
  }
});
Object.defineProperty(exports, "NumberDecrementStepper", {
  enumerable: true,
  get: function () {
    return _composites.NumberDecrementStepper;
  }
});
Object.defineProperty(exports, "Collapse", {
  enumerable: true,
  get: function () {
    return _composites.Collapse;
  }
});
Object.defineProperty(exports, "CircularProgress", {
  enumerable: true,
  get: function () {
    return _composites.CircularProgress;
  }
});
Object.defineProperty(exports, "Modal", {
  enumerable: true,
  get: function () {
    return _composites.Modal;
  }
});
Object.defineProperty(exports, "Popover", {
  enumerable: true,
  get: function () {
    return _composites.Popover;
  }
});
Object.defineProperty(exports, "AlertDialog", {
  enumerable: true,
  get: function () {
    return _composites.AlertDialog;
  }
});
Object.defineProperty(exports, "useToast", {
  enumerable: true,
  get: function () {
    return _composites.useToast;
  }
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function () {
    return _composites.Menu;
  }
});
Object.defineProperty(exports, "SimpleGrid", {
  enumerable: true,
  get: function () {
    return _composites.SimpleGrid;
  }
});
Object.defineProperty(exports, "Tabs", {
  enumerable: true,
  get: function () {
    return _composites.Tabs;
  }
});
Object.defineProperty(exports, "Actionsheet", {
  enumerable: true,
  get: function () {
    return _composites.Actionsheet;
  }
});
Object.defineProperty(exports, "Snackbar", {
  enumerable: true,
  get: function () {
    return _composites.Snackbar;
  }
});
Object.defineProperty(exports, "useSnackbar", {
  enumerable: true,
  get: function () {
    return _composites.useSnackbar;
  }
});
Object.defineProperty(exports, "Fab", {
  enumerable: true,
  get: function () {
    return _composites.Fab;
  }
});
Object.defineProperty(exports, "Typeahead", {
  enumerable: true,
  get: function () {
    return _composites.Typeahead;
  }
});
Object.defineProperty(exports, "useTypeahead", {
  enumerable: true,
  get: function () {
    return _composites.useTypeahead;
  }
});
Object.defineProperty(exports, "View", {
  enumerable: true,
  get: function () {
    return _primitives.View;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function () {
    return _primitives.Text;
  }
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function () {
    return _primitives.Checkbox;
  }
});
Object.defineProperty(exports, "Radio", {
  enumerable: true,
  get: function () {
    return _primitives.Radio;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function () {
    return _primitives.Button;
  }
});
Object.defineProperty(exports, "ButtonGroup", {
  enumerable: true,
  get: function () {
    return _primitives.ButtonGroup;
  }
});
Object.defineProperty(exports, "Column", {
  enumerable: true,
  get: function () {
    return _primitives.Column;
  }
});
Object.defineProperty(exports, "Row", {
  enumerable: true,
  get: function () {
    return _primitives.Row;
  }
});
Object.defineProperty(exports, "Box", {
  enumerable: true,
  get: function () {
    return _primitives.Box;
  }
});
Object.defineProperty(exports, "Stack", {
  enumerable: true,
  get: function () {
    return _primitives.Stack;
  }
});
Object.defineProperty(exports, "VStack", {
  enumerable: true,
  get: function () {
    return _primitives.VStack;
  }
});
Object.defineProperty(exports, "HStack", {
  enumerable: true,
  get: function () {
    return _primitives.HStack;
  }
});
Object.defineProperty(exports, "ZStack", {
  enumerable: true,
  get: function () {
    return _primitives.ZStack;
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function () {
    return _primitives.Slider;
  }
});
Object.defineProperty(exports, "Icon", {
  enumerable: true,
  get: function () {
    return _primitives.Icon;
  }
});
Object.defineProperty(exports, "createIcon", {
  enumerable: true,
  get: function () {
    return _primitives.createIcon;
  }
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function () {
    return _primitives.Input;
  }
});
Object.defineProperty(exports, "InputGroup", {
  enumerable: true,
  get: function () {
    return _primitives.InputGroup;
  }
});
Object.defineProperty(exports, "InputLeftAddon", {
  enumerable: true,
  get: function () {
    return _primitives.InputLeftAddon;
  }
});
Object.defineProperty(exports, "InputRightAddon", {
  enumerable: true,
  get: function () {
    return _primitives.InputRightAddon;
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () {
    return _primitives.Image;
  }
});
Object.defineProperty(exports, "Spinner", {
  enumerable: true,
  get: function () {
    return _primitives.Spinner;
  }
});
Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function () {
    return _primitives.Heading;
  }
});
Object.defineProperty(exports, "Flex", {
  enumerable: true,
  get: function () {
    return _primitives.Flex;
  }
});
Object.defineProperty(exports, "Switch", {
  enumerable: true,
  get: function () {
    return _primitives.Switch;
  }
});
Object.defineProperty(exports, "TextArea", {
  enumerable: true,
  get: function () {
    return _primitives.TextArea;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function () {
    return _primitives.Link;
  }
});
Object.defineProperty(exports, "Spacer", {
  enumerable: true,
  get: function () {
    return _primitives.Spacer;
  }
});
Object.defineProperty(exports, "List", {
  enumerable: true,
  get: function () {
    return _primitives.List;
  }
});
Object.defineProperty(exports, "Select", {
  enumerable: true,
  get: function () {
    return _primitives.Select;
  }
});
exports.default = void 0;

var _composites = require("./components/composites");

var _primitives = require("./components/primitives");

var _theme = require("./theme");

Object.keys(_theme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _theme[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _theme[key];
    }
  });
});

var _core = require("./core");

Object.keys(_core).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _core[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _core[key];
    }
  });
});

var _hooks = require("./hooks");

Object.keys(_hooks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hooks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _hooks[key];
    }
  });
});

var _factory = require("./factory");

Object.keys(_factory).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _factory[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _factory[key];
    }
  });
});
var _default = {
  multiply(a, b) {
    return Promise.resolve(a * b);
  }

};
exports.default = _default;
