// native-base/lib/commonjs/components/composites/Tabs/Tab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _Box = _interopRequireDefault(require("../../primitives/Box"));

var _Context = require("./Context");

var _utils = require("../../../theme/tools/utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Tab = ({
  index,
  children,
  isDisabled,
  style,
  _active,
  _disabled,
  ...props
}) => {
  const newProps = (0, _utils.omitUndefined)(props);

  const {
    activeIndex,
    inactiveTabStyle,
    activeTabStyle,
    changeHandler
  } = _react.default.useContext(_Context.TabsContext);

  const isOpen = activeIndex === index;
  const tabStyle = isOpen ? activeTabStyle : inactiveTabStyle;

  const pressHandler = () => {
    changeHandler && changeHandler(index);
  };

  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    activeOpacity: 1,
    disabled: isDisabled,
    onPress: () => pressHandler(),
    accessible: true,
    accessibilityRole: "radio"
  }, /*#__PURE__*/_react.default.createElement(_Box.default, _extends({}, tabStyle, newProps, {
    style: [style, isOpen && _active, isDisabled && _disabled]
  }), children));
};

var _default = /*#__PURE__*/_react.default.memo(Tab);

exports.default = _default;
