// native-base/lib/commonjs/theme/components/avatar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _tools = require("./../tools");

const baseStyle = props => {
  const {
    name,
    ...colorModeProps
  } = props;
  const bg = name ? (0, _tools.randomColor)({
    string: (0, _tools.getRandomString)(5) + name
  }) : 'gray.400';
  const borderColor = (0, _tools.mode)('gray.800', 'white')(colorModeProps);
  return {
    bg,
    borderColor,
    position: 'relative',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 'full',
    _text: {
      fontWeight: 600
    }
  };
};

function getSize(size, fontSize) {
  return {
    width: size,
    height: size,
    fontSize: fontSize
  };
}

const sizes = {
  '2xs': getSize('4', 'xs'),
  'xs': getSize('6', 'sm'),
  'sm': getSize('8', 'md'),
  'md': getSize('12', 'lg'),
  'lg': getSize('16', 'xl'),
  'xl': getSize('24', '2xl'),
  '2xl': getSize('32', '3xl'),
  'full': getSize('100%', '4xl')
};
const defaultProps = {
  size: 'md'
};
var _default = {
  baseStyle,
  sizes,
  defaultProps
};
exports.default = _default;
