// native-base/lib/commonjs/theme/components/badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _tools = require("../tools");

const baseStyle = {
  px: 1,
  py: 1,
  _text: {
    fontSize: 'xs',
    textTransform: 'uppercase',
    fontWeight: 'bold'
  },
  borderRadius: 'lg'
};

function variantSolid(props) {
  let colorScheme = (0, _tools.getColorScheme)(props);
  return {
    bg: (0, _tools.mode)("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props),
    borderWidth: 1,
    borderColor: 'transparent'
  };
}

function variantSubtle(props) {
  let colorScheme = (0, _tools.getColorScheme)(props);
  return {
    bg: (0, _tools.mode)("".concat(colorScheme, ".100"), "".concat(colorScheme, ".500"))(props),
    _text: {
      color: (0, _tools.mode)("".concat(colorScheme, ".700"), "".concat(colorScheme, ".100"))(props)
    },
    borderWidth: 1,
    borderColor: 'transparent'
  };
}

function variantOutline(props) {
  let colorScheme = (0, _tools.getColorScheme)(props);
  return {
    borderColor: (0, _tools.mode)("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props),
    _text: {
      color: (0, _tools.mode)("".concat(colorScheme, ".500"), "".concat(colorScheme, ".100"))(props)
    },
    borderWidth: 1
  };
}

const variants = {
  solid: variantSolid,
  subtle: variantSubtle,
  outline: variantOutline
};
const defaultProps = {
  variant: 'subtle',
  colorScheme: 'gray'
};
var _default = {
  baseStyle,
  variants,
  defaultProps
};
exports.default = _default;
