// native-base/lib/commonjs/theme/components/menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MenuItem = exports.MenuGroup = exports.default = void 0;

var _tools = require("./../tools");

function baseStyle(props) {
  return {
    bg: (0, _tools.mode)("#fff", "gray.700")(props),
    // change this to match theme
    borderRadius: 'md',
    borderWidth: '1px',
    borderColor: 'gray.300'
  };
}

const defaultProps = {
  shadow: 1
};
var _default = {
  defaultProps,
  baseStyle
};
exports.default = _default;
const MenuGroup = {
  baseStyle: {
    _title: {
      fontSize: 14,
      fontWeight: 'semibold'
    },
    p: 2
  }
};
exports.MenuGroup = MenuGroup;
const MenuItem = {
  baseStyle: {
    _text: {
      fontSize: 14,
      fontWeight: 'normal',
      textAlign: 'left'
    },
    py: 2,
    px: 3,
    maxWidth: 64,
    minWidth: 32
  },
  defaultProps: {
    isDisabled: false,
    disabledTextColor: '#bdbdbd',
    // change this to match theme
    underlayColor: '#E8EDFB',
    _disabled: {
      bg: 'gray.100',
      // color: 'gray.500',
      underlayColor: '#f4f4f5'
    }
  }
};
exports.MenuItem = MenuItem;
