// native-base/lib/commonjs/theme/components/modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalOverlay = exports.ModalFooter = exports.ModalBody = exports.ModalHeader = exports.ModalCloseButton = exports.ModalContent = exports.Modal = void 0;

var _tools = require("../tools");

const sizes = {
  xs: {
    contentSize: {
      width: '40%'
    }
  },
  sm: {
    contentSize: {
      width: '48%'
    }
  },
  md: {
    contentSize: {
      width: '60%'
    }
  },
  lg: {
    contentSize: {
      width: '75%'
    }
  },
  xl: {
    contentSize: {
      width: '90%'
    }
  },
  full: {
    contentSize: {
      width: '100%'
    }
  }
};
const Modal = {
  baseStyle: {
    width: '100%',
    height: '100%' // border: 1,

  },
  sizes,
  defaultProps: {
    size: 'md',
    closeOnOverlayClick: true
  }
};
exports.Modal = Modal;
const ModalContent = {
  baseStyle: props => {
    return {
      bg: (0, _tools.mode)('gray.100', 'gray.700')(props),
      p: 8,
      rounded: 'lg'
    };
  }
};
exports.ModalContent = ModalContent;
const ModalCloseButton = {
  baseStyle: {
    position: 'absolute',
    right: 8,
    top: 8,
    zIndex: 1
  },
  defaultProps: {
    size: 8
  }
};
exports.ModalCloseButton = ModalCloseButton;
const ModalHeader = {
  baseStyle: {
    pb: 3
  }
};
exports.ModalHeader = ModalHeader;
const ModalBody = {
  baseStyle: {
    mb: 3,
    fontSize: '2xl'
  }
};
exports.ModalBody = ModalBody;
const ModalFooter = {
  baseStyle: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    flexWrap: 'wrap'
  }
};
exports.ModalFooter = ModalFooter;
const ModalOverlay = {
  baseStyle: {
    position: 'absolute',
    left: 0,
    top: 0,
    opacity: 0.5,
    right: 0,
    bottom: 0
  }
};
exports.ModalOverlay = ModalOverlay;
