// native-base/lib/commonjs/theme/components/select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectItem = exports.Select = void 0;

var _colors = require("../tools/colors");

// styled
const styled = props => {
  return {
    variant: 'styled',
    pl: 2,
    pr: 0,
    border: 1,
    borderColor: (0, _colors.mode)('gray.600', 'gray.400')(props),
    _isDisabled: {
      opacity: 0.4
    },
    _isInvalid: {
      borderColor: (0, _colors.mode)('danger.600', 'danger.200')(props)
    },
    _hover: {
      borderColor: 'default.500'
    }
  };
}; // native


const native = props => {
  return {
    variant: 'native',
    androidIconColor: (0, _colors.mode)('darkText', 'lightText')(props),
    color: (0, _colors.mode)('darkText', 'lightText')(props),
    _android: {
      p: 2
    },
    _ios: {},
    _web: {
      bg: 'transparent',
      borderRadius: 'md',
      borderColor: 'default.500',
      p: 3
    },
    _hover: {
      borderColor: 'default.500'
    }
  };
};

const variants = {
  styled: styled,
  native: native
}; // Select

const Select = {
  variants,
  defaultProps: {
    variant: 'native'
  }
}; // SelectIcon - only for styled variant

exports.Select = Select;
const SelectItem = {
  baseStyle: {
    p: 1,
    px: 2,
    borderRadius: 0,
    minH: 0
  }
};
exports.SelectItem = SelectItem;
