// native-base/lib/commonjs/theme/components/slider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SliderThumb = exports.default = void 0;

var _tools = require("../tools");

const baseStyle = props => {
  const simplifiedColorScheme = (0, _tools.getColorScheme)(props);
  return {
    activeColor: (0, _tools.mode)("".concat(simplifiedColorScheme, ".500"), "".concat(simplifiedColorScheme, ".200"))(props),
    trackColor: (0, _tools.mode)('gray.200', 'gray.700')(props)
  };
};

const sizes = {
  // sizes mentioned here are used in pxiles.
  lg: {
    thumbSize: 10,
    sliderSize: 9
  },
  md: {
    thumbSize: 8,
    sliderSize: 8
  },
  sm: {
    thumbSize: 6,
    sliderSize: 6
  }
};
const defaultProps = {
  colorScheme: 'default',
  size: 'md',
  min: 0,
  max: 100,
  step: 1
};
var _default = {
  baseStyle,
  sizes,
  defaultProps
}; // SliderThumb

exports.default = _default;
const sliderThumbDefaultProps = {
  shadow: 2
};
const SliderThumb = {
  defaultProps: sliderThumbDefaultProps
};
exports.SliderThumb = SliderThumb;
