// native-base/lib/commonjs/theme/components/switch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _tools = require("../tools");

const baseStyle = props => {
  let {
    onTrackColor,
    offTrackColor,
    onThumbColor,
    offThumbColor
  } = props;
  return {
    offTrackColor: offTrackColor !== null && offTrackColor !== void 0 ? offTrackColor : (0, _tools.mode)('gray.100', 'gray.900')(props),
    onTrackColor: onTrackColor !== null && onTrackColor !== void 0 ? onTrackColor : (0, _tools.mode)('green.300', 'green.700')(props),
    onThumbColor: onThumbColor !== null && onThumbColor !== void 0 ? onThumbColor : (0, _tools.mode)('white', 'black')(props),
    offThumbColor: offThumbColor !== null && offThumbColor !== void 0 ? offThumbColor : (0, _tools.mode)('white', 'black')(props)
  };
};

const sizes = {
  sm: {
    transform: [{
      scale: 0.75
    }]
  },
  md: {},
  lg: {
    transform: [{
      scale: 1.25
    }],
    margin: 1
  }
};
const defaultProps = {
  size: 'md'
};
var _default = {
  baseStyle,
  sizes,
  defaultProps
};
exports.default = _default;
