// native-base/lib/commonjs/theme/components/tabs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _tools = require("../tools");

const baseStyle = {
  activeTabStyle: {
    marginBottom: '-2px'
  },
  inactiveTabStyle: {
    marginBottom: '-2px'
  }
};
const sizes = {
  sm: {
    activeTabStyle: {
      fontSize: 'sm',
      p: 1
    },
    inactiveTabStyle: {
      fontSize: 'sm',
      p: 1
    }
  },
  md: {
    activeTabStyle: {
      fontSize: 'md',
      p: 2
    },
    inactiveTabStyle: {
      fontSize: 'md',
      p: 2
    }
  },
  lg: {
    activeTabStyle: {
      fontSize: 'lg',
      p: 3
    },
    inactiveTabStyle: {
      fontSize: 'lg',
      p: 3
    }
  }
};

function variantLine(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      borderColor: activeColor,
      borderBottomWidth: 2
    },
    inactiveTabStyle: {
      borderColor: 'transparent',
      borderBottomWidth: 2
    },
    tabBarStyle: {
      borderBottomWidth: 2,
      borderColor: 'muted.200'
    }
  };
}

function variantEnclosed(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      borderTopWidth: 2,
      borderLeftWidth: 2,
      borderRightWidth: 2,
      borderColor: activeColor
    },
    inactiveTabStyle: {
      borderColor: 'muted.200',
      borderBottomWidth: 2
    },
    tabBarStyle: {}
  };
}

function variantEnclosedColor(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      borderTopWidth: 2,
      borderLeftWidth: 2,
      borderRightWidth: 2,
      borderColor: 'muted.200',
      color: activeColor
    },
    inactiveTabStyle: {
      bg: (0, _tools.transparentize)('muted.200', 0.1)(props.theme),
      borderColor: 'muted.200',
      borderBottomWidth: 2
    },
    tabBarStyle: {}
  };
}

function varientSoftRounded(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      borderRadius: 'full',
      bg: (0, _tools.transparentize)(activeColor, 0.7)(props.theme),
      color: 'white'
    },
    inactiveTabStyle: {},
    tabBarStyle: {}
  };
}

function varientSolidRounded(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      borderRadius: 'full',
      bg: activeColor,
      color: 'white'
    },
    inactiveTabStyle: {},
    tabBarStyle: {}
  };
}

function varientSolidBox(props) {
  const activeColor = (0, _tools.getColorFormColorScheme)(props);
  return {
    activeTabStyle: {
      bg: activeColor,
      color: 'white'
    },
    inactiveTabStyle: {},
    tabBarStyle: {}
  };
}

const variants = {
  'line': variantLine,
  'enclosed': variantEnclosed,
  'enclosed-colored': variantEnclosedColor,
  'soft-rounded': varientSoftRounded,
  'solid-rounded': varientSolidRounded,
  'solid-box': varientSolidBox
};
const defaultProps = {
  size: 'md',
  variant: 'line',
  colorScheme: 'default'
};
var _default = {
  baseStyle,
  variants,
  sizes,
  defaultProps
};
exports.default = _default;
