// native-base/lib/commonjs/components/primitives/Select/SelectItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Item = void 0;

var _react = _interopRequireDefault(require("react"));

var _core = require("../../../core");

var _Button = _interopRequireDefault(require("../Button"));

var _Text = _interopRequireDefault(require("../Text"));

var _picker = require("@react-native-picker/picker");

var _hooks = require("../../../hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Item = ({
  isDisabled,
  label,
  value,
  _label,
  style,
  ...props
}) => {
  const {
    parentComponentConfig
  } = (0, _core.usePopover)();
  const { ...newProps
  } = (0, _hooks.useThemeProps)('SelectItem', props);

  if ((parentComponentConfig === null || parentComponentConfig === void 0 ? void 0 : parentComponentConfig.variant) === 'styled') {
    const {
      selectedValue,
      closeMenu,
      selectedItemBg,
      _selectedItem,
      onValueChange,
      itemsList,
      _item,
      width
    } = parentComponentConfig;
    let currentIndex = -1;
    itemsList.forEach((item, index) => {
      if (item.value === value) {
        currentIndex = index;
      }
    });
    let textProps = { ..._item,
      ..._label
    };

    if (selectedValue === value) {
      textProps = { ..._selectedItem
      };
    }

    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      width: width !== null && width !== void 0 ? width : 150
    }, newProps, {
      bg: selectedValue === value ? selectedItemBg : undefined,
      justifyContent: "flex-start",
      style: style,
      onPress: () => {
        if (!isDisabled) {
          onValueChange(value, currentIndex);
          closeMenu && closeMenu();
        }
      }
    }), /*#__PURE__*/_react.default.createElement(_Text.default, _extends({
      fontSize: "sm",
      key: "select-item-".concat(value)
    }, textProps), label));
  } else {
    return /*#__PURE__*/_react.default.createElement(_picker.Picker.Item, {
      label: label,
      value: value
    });
  }
};

exports.Item = Item;

var _default = /*#__PURE__*/_react.default.memo(Item);

exports.default = _default;
