// native-base/lib/module/components/primitives/Select/SelectItem.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { usePopover } from '../../../core';
import Button from '../Button';
import Text from '../Text';
import { Picker as RNPicker } from '@react-native-picker/picker';
import { useThemeProps } from '../../../hooks';
export const Item = ({
  isDisabled,
  label,
  value,
  _label,
  style,
  ...props
}) => {
  const {
    parentComponentConfig
  } = usePopover();
  const { ...newProps
  } = useThemeProps('SelectItem', props);

  if ((parentComponentConfig === null || parentComponentConfig === void 0 ? void 0 : parentComponentConfig.variant) === 'styled') {
    const {
      selectedValue,
      closeMenu,
      selectedItemBg,
      _selectedItem,
      onValueChange,
      itemsList,
      _item,
      width
    } = parentComponentConfig;
    let currentIndex = -1;
    itemsList.forEach((item, index) => {
      if (item.value === value) {
        currentIndex = index;
      }
    });
    let textProps = { ..._item,
      ..._label
    };

    if (selectedValue === value) {
      textProps = { ..._selectedItem
      };
    }

    return /*#__PURE__*/React.createElement(Button, _extends({
      width: width !== null && width !== void 0 ? width : 150
    }, newProps, {
      bg: selectedValue === value ? selectedItemBg : undefined,
      justifyContent: "flex-start",
      style: style,
      onPress: () => {
        if (!isDisabled) {
          onValueChange(value, currentIndex);
          closeMenu && closeMenu();
        }
      }
    }), /*#__PURE__*/React.createElement(Text, _extends({
      fontSize: "sm",
      key: "select-item-".concat(value)
    }, textProps), label));
  } else {
    return /*#__PURE__*/React.createElement(RNPicker.Item, {
      label: label,
      value: value
    });
  }
};
export default /*#__PURE__*/React.memo(Item);
