// native-base/lib/commonjs/components/composites/Popover/PopoverCloseButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _View = _interopRequireDefault(require("../../primitives/View"));

var _CloseButton = _interopRequireDefault(require("../CloseButton"));

var _popover = require("./popover");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const PopoverCloseButton = props => {
  const {
    PopoverRef,
    newProps: {
      popoverCloseButtonProps
    }
  } = _react.default.useContext(_popover.PopoverContext);

  return /*#__PURE__*/_react.default.createElement(_View.default, {
    position: "absolute",
    right: 0,
    top: 0,
    zIndex: 1
  }, /*#__PURE__*/_react.default.createElement(_CloseButton.default, _extends({}, popoverCloseButtonProps, props, {
    onPress: () => PopoverRef === null || PopoverRef === void 0 ? void 0 : PopoverRef.current.toggleTooltip()
  })));
};

var _default = /*#__PURE__*/_react.default.memo(PopoverCloseButton);

exports.default = _default;
