// native-base/lib/commonjs/components/composites/Popover/Triangle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _View = _interopRequireDefault(require("../../primitives/View"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Triangle = ({
  style,
  isDown,
  borderBottomColor
}) => /*#__PURE__*/_react.default.createElement(_View.default, {
  borderBottomColor: borderBottomColor,
  style: [styles.triangle, style, isDown ? styles.down : {}]
});

const styles = _reactNative.StyleSheet.create({
  down: {
    transform: [{
      rotate: '180deg'
    }]
  },
  triangle: {
    width: 0,
    height: 0,
    backgroundColor: 'transparent',
    borderStyle: 'solid',
    borderLeftWidth: 8,
    borderRightWidth: 8,
    borderBottomWidth: 15,
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent'
  }
});

var _default = Triangle;
exports.default = _default;
