// native-base/lib/commonjs/components/primitives/List/ListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("../Box"));

var _hooks = require("../../../hooks");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const ListItem = /*#__PURE__*/_react.default.memo(({
  children,
  ...props
}) => {
  const {
    index,
    start,
    unordered,
    ul,
    ordered,
    ol,
    _text,
    borderTopWidth,
    ...newProps
  } = (0, _hooks.useThemeProps)('ListItem', props);
  return /*#__PURE__*/_react.default.createElement(_Box.default, _extends({
    flexDirection: "row",
    alignItems: "center"
  }, newProps, {
    borderTopWidth: index ? borderTopWidth : 0
  }), /*#__PURE__*/_react.default.createElement(_Box.default, {
    flexDirection: "row",
    alignItems: "center",
    pl: 2
  }, ul || unordered ?
  /*#__PURE__*/
  //Adding disc in front of ListItem
  _react.default.createElement(_Box.default, {
    style: {
      transform: [{
        scale: 1.5
      }]
    },
    mr: 2,
    _text: {
      fontWeight: 'bold',
      ..._text
    }
  }, "\u2022") : null, ol || ordered ?
  /*#__PURE__*/
  //Adding index number in front of ListItem
  _react.default.createElement(_Box.default, {
    mr: 2,
    _text: {
      fontWeight: 'bold',
      ..._text
    }
  }, index + start + '.') : null), /*#__PURE__*/_react.default.createElement(_Box.default, {
    flexDirection: "row",
    alignItems: "center",
    _text: { ..._text
    }
  }, children));
});

var _default = /*#__PURE__*/_react.default.memo(ListItem);

exports.default = _default;
