// native-base/lib/commonjs/components/composites/Alert/Alert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("../../primitives/Box"));

var _hooks = require("../../../hooks");

var _Context = require("./Context");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Alert = ({
  children,
  _text,
  action,
  ...props
}, ref) => {
  const {
    status,
    variant,
    iconColor,
    textColor,
    ...newProps
  } = (0, _hooks.useThemeProps)('Alert', props);
  return /*#__PURE__*/_react.default.createElement(_Context.AlertContext.Provider, {
    value: {
      status,
      variant,
      iconColor,
      textColor
    }
  }, /*#__PURE__*/_react.default.createElement(_Box.default, _extends({
    minW: "100%",
    maxW: "100%",
    position: "relative",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start",
    flexWrap: "wrap",
    _text: _text
  }, newProps, {
    ref: ref
  }), children, action ? /*#__PURE__*/_react.default.createElement(_Box.default, {
    mr: 0
  }, action) : null));
};

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(Alert));

exports.default = _default;
