// native-base/lib/module/components/composites/Skeleton/Skeleton.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { Animated, Platform, View } from 'react-native';
import { useThemeProps } from '../../../hooks';
import { canUseDom } from '../../../utils';
import Box from '../../primitives/Box';

const Skeleton = (allProps, ref) => {
  var _props$transform;

  const isDomUsable = canUseDom();
  const {
    variant,
    children,
    ...props
  } = allProps;
  const newProps = useThemeProps('Skeleton', props);
  const {
    style,
    skeletonColor,
    baseColor
  } = newProps;
  const blinkAnim = React.useRef(new Animated.Value(0)).current;
  React.useEffect(() => {
    if (isDomUsable) {
      const blink = Animated.sequence([Animated.timing(blinkAnim, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: Platform.OS !== 'web'
      }), Animated.timing(blinkAnim, {
        toValue: 0,
        duration: 1000,
        useNativeDriver: Platform.OS !== 'web'
      })]);
      Animated.loop(blink).start();
    }
  }, [blinkAnim, isDomUsable]);
  const skeletonStyle = {
    skeleton: {
      position: 'absolute',
      top: 0,
      bottom: 0,
      height: '100%',
      width: '100%',
      borderRadius: variant === 'circle' ? 999 : 3,
      backgroundColor: skeletonColor,
      opacity: blinkAnim // Bind opacity to animated value

    }
  };
  return /*#__PURE__*/React.createElement(Box, _extends({
    style: [style, {
      transform: ((_props$transform = props.transform) !== null && _props$transform !== void 0 ? _props$transform : variant === 'text') ? [{
        scaleY: 0.6
      }] : undefined
    }],
    borderRadius: variant === 'circle' ? 999 : 3,
    bg: baseColor
  }, props, {
    ref: ref
  }), /*#__PURE__*/React.createElement(Animated.View, {
    style: skeletonStyle.skeleton
  }), children ? /*#__PURE__*/React.createElement(View, {
    style: {
      opacity: 0
    }
  }, children) : null);
};

export default /*#__PURE__*/React.memo( /*#__PURE__*/React.forwardRef(Skeleton));
