// native-base/lib/module/components/composites/NumberInput/NumberInputStepper.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import { VStack, Box, Icon } from '../../primitives';
import { useThemeProps } from '../../../hooks';
import { NumberInputContext } from './Context';
export const NBStepper = ({
  children,
  ...props
}) => {
  const {
    style,
    isIncrement,
    disablitityCheck,
    _active,
    _disabled,
    isDisabled,
    ariaLabel,
    pressHandler,
    iconColor,
    ...newProps
  } = useThemeProps('NumberInputStepper', props);
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    activeOpacity: 1,
    disabled: disablitityCheck || isDisabled,
    onPress: pressHandler,
    accessible: true,
    accessibilityLabel: ariaLabel
  }, /*#__PURE__*/React.createElement(Box, _extends({}, newProps, _active, disablitityCheck || isDisabled ? _disabled : {}, {
    borderColor: "transparent",
    style: style,
    opacity: disablitityCheck || isDisabled ? 0.4 : 1
  }, Platform.OS === 'web' ? {
    disabled: disablitityCheck || isDisabled,
    cursor: disablitityCheck || isDisabled ? 'not-allowed' : 'auto'
  } : {}), children || isIncrement ? /*#__PURE__*/React.createElement(Icon, {
    name: "arrow-drop-up",
    type: "MaterialIcons",
    color: iconColor
  }) : /*#__PURE__*/React.createElement(Icon, {
    name: "arrow-drop-down",
    type: "MaterialIcons",
    color: iconColor
  })));
};

const NumberInputStepper = ({
  children,
  ...props
}) => {
  const {
    numberInputStepper,
    setNumberInputStepper
  } = React.useContext(NumberInputContext);
  React.useEffect(() => {
    !numberInputStepper && setNumberInputStepper( /*#__PURE__*/React.createElement(VStack, props, children));
  }, [numberInputStepper, setNumberInputStepper, props, children]);
  return null;
};

export default /*#__PURE__*/React.memo(NumberInputStepper);
