// native-base/lib/commonjs/components/primitives/Text/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _native = _interopRequireDefault(require("styled-components/native"));

var _styledSystem = require("styled-system");

var _useThemeProps = require("../../../hooks/useThemeProps");

var _customProps = require("../../../utils/customProps");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const StyledText = (0, _native.default)(_reactNative.Text)(_styledSystem.color, _styledSystem.space, _styledSystem.position, _styledSystem.layout, _styledSystem.flexbox, _styledSystem.border, _styledSystem.typography, _styledSystem.position, _customProps.customPosition, _customProps.customBorder, _customProps.customBackground, _customProps.customOutline, _customProps.customShadow, _customProps.customExtra, _customProps.customLayout, _customProps.customTypography);

const Text = ({
  children,
  style,
  isTruncated,
  noOfLines,
  bold,
  italic,
  sub,
  highlight,
  underline,
  strikeThrough,
  ...props
}, ref) => {
  const newProps = (0, _useThemeProps.useThemeProps)('Text', props);
  return /*#__PURE__*/_react.default.createElement(StyledText, _extends({}, newProps, {
    numberOfLines: noOfLines ? noOfLines : isTruncated ? 1 : 999,
    fontWeight: bold ? 'bold' : newProps.fontWeight,
    fontStyle: italic ? 'italic' : newProps.fontStyle,
    bg: highlight ? 'yellow.200' : newProps.bg,
    textDecorationLine: underline ? 'underline' : strikeThrough ? 'line-through' : newProps.textDecorationLine,
    fontSize: sub ? 10 : newProps.fontSize,
    style: style,
    ref: ref
  }), children);
};

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(Text));

exports.default = _default;
