// native-base/lib/commonjs/components/composites/CloseButton/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconButton = _interopRequireDefault(require("../../composites/IconButton"));

var _Icon = _interopRequireDefault(require("../../primitives/Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const CloseButton = (props, ref) => {
  const iconSize = {
    xs: 6,
    sm: 8,
    md: 10,
    lg: 12
  };
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, _extends({}, props, {
    ref: ref,
    icon: /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: 'close',
      size: props.size ? iconSize[props.size] : 8
    })
  }));
};

var _default = /*#__PURE__*/_react.default.memo( /*#__PURE__*/_react.default.forwardRef(CloseButton));

exports.default = _default;
