// native-base/lib/module/core/color-mode/hooks.js
import React, { useState, useEffect } from 'react';
import { HybridContext } from './../hybrid-overlay/Context';
export const useColorMode = () => {
  const {
    colorMode: context
  } = React.useContext(HybridContext);

  if (context === undefined) {
    throw new Error('useColorMode must be used within a NativeBaseProvider');
  }

  return context;
};
export function useColorModeValue(light, dark) {
  const {
    colorMode
  } = useColorMode();
  return colorMode === 'dark' ? dark : light;
}
export function useModeManager(initialColorMode, colorModeManager) {
  const [colorMode, setRawMode] = useState(initialColorMode);

  async function setColorMode(val) {
    if (colorModeManager) {
      await colorModeManager.set(val);
    }

    setRawMode(val);
  } // For initial setting initial color mode from storage


  useEffect(() => {
    if (colorModeManager) {
      (async function getMode() {
        let value = await colorModeManager.get(initialColorMode);

        if (value && value !== colorMode) {
          setRawMode(value);
        }
      })();
    }
  }, [colorMode, initialColorMode, colorModeManager]);
  return {
    colorMode,
    setColorMode
  };
}
